fastify.js

```diff
@@ -882,6 +887,36 @@ function build (options) {
 
     return middie
   }
+
+  function getTrustProxyFn () {
+    const tp = options.trustProxy
+    if (typeof tp === 'function') {
+      return tp
+    }
+    if (tp === true) {
+      // Support plain true/false
+      return function () { return true }
+    }
+    if (typeof tp === 'number') {
+      // Support trusting hop count
+      return function (a, i) { return i < tp }
+    }
+    if (typeof tp === 'string') {
+      // Support comma-separated tps
+      const vals = tp.split(',').map(it => it.trim())
+      return proxyAddr.compile(vals)
+    }
+    return proxyAddr.compile(tp || [])
+  }
+
+  function _handleTrustProxy (req) {
+    const proxyFn = getTrustProxyFn()
```

Comment:
can you move this out of this function? This is likely to be expensive to create for every request.

Reply:
done